src/main/java/seedu/tr4cker/logic/commands/CountdownCommand.java

```diff
@@ -95,35 +109,65 @@ public CountdownCommand(Index index, boolean isDeleteCountdown) {
         this.eventDate = null;
         this.index = index;
         this.isDeleteCountdown = isDeleteCountdown;
+        this.numDays = -1;
+    }
+
+    /**
+     * Constructor for CountdownCommand when user wants to list number of countdowns in {@code days}.
+     */
+    public CountdownCommand(int numDays) {
+        this.eventName = null;
+        this.eventDate = null;
+        this.index = null;
+        this.isDeleteCountdown = false;
+        this.numDays = numDays;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Event> eventList = model.getFilteredEventList();
 
-        if (Objects.isNull(eventName) && Objects.isNull(eventDate) && Objects.isNull(index)) {
+        if (Objects.isNull(eventName) && Objects.isNull(eventDate) && Objects.isNull(index) && numDays == -1) {
             model.updateFilteredEventList(PREDICATE_SHOW_ALL_EVENTS);
             return new CommandResult(MESSAGE_SWITCH_TAB_SUCCESS, true);
         }
 
-        if (Objects.isNull(eventName) && Objects.isNull(eventDate)) {
+        if (Objects.isNull(eventName) && Objects.isNull(eventDate) && numDays == -1) {
             if (isDeleteCountdown) {
                 return executeCountdownDelete(model, eventList);
             } else {
                 return executeCountdownAddFromTask(model);
             }
         }
 
-        if (Objects.isNull(index)) {
+        if (Objects.isNull(index) && numDays == -1) {
             return executeCountdownAddNew(model);
         }
 
+        if (Objects.isNull(eventName) && Objects.isNull(eventDate) && Objects.isNull(index)) {
```

Comment:
Perhaps you can use else-ifs to make this method more readable since only one of the clauses will be executed?